genesyscloud/outbound_digitalruleset/resource_genesyscloud_outbound_digitalruleset_schema.go

```diff
@@ -0,0 +1,550 @@
+				Optional:    true,
+				Type:        schema.TypeSet,
+				MaxItems:    1,
+				Elem:        markContactUncontactableActionSettingsResource,
+			},
+			`mark_contact_address_uncontactable_action_settings`: {
+				Description:      `The settings for an 'mark contact address uncontactable' action.`,
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				DiffSuppressFunc: util.SuppressEquivalentJsonDiffs,
+			},
+			`set_content_template_action_settings`: {
+				Description: `The settings for a 'Set content template' action.`,
+				Optional:    true,
+				Type:        schema.TypeSet,
+				MaxItems:    1,
+				Elem:        setContentTemplateActionSettingsResource,
+			},
+			`set_sms_phone_number_action_settings`: {
+				Description: `The settings for a 'set sms phone number' action.`,
+				Optional:    true,
+				Type:        schema.TypeSet,
+				MaxItems:    1,
+				Elem:        setSmsPhoneNumberActionSettingsResource,
+			},
+		},
+	}
+
+	digitalRuleResource = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			`name`: {
+				Description: `The name of the rule.`,
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			`order`: {
+				Description: `The ranked order of the rule. Rules are processed from lowest number to highest.`,
+				Required:    true,
+				Type:        schema.TypeInt,
+			},
+			`category`: {
+				Description:  `The category of the rule.`,
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{"PreContact", "PostContact"}, false),
+			},
+			`conditions`: {
+				Description: `A list of conditions to evaluate. All of the Conditions must evaluate to true to trigger the actions.`,
+				Required:    true,
+				Type:        schema.TypeList,
+				Elem:        digitalConditionResource,
+			},
+			`actions`: {
+				Description: `The list of actions to be taken if all conditions are true.`,
+				Required:    true,
+				Type:        schema.TypeList,
+				Elem:        digitalActionResource,
+			},
+		},
+	}
+)
+
+// ResourceOutboundDigitalruleset registers the genesyscloud_outbound_digitalruleset resource with Terraform
+func ResourceOutboundDigitalruleset() *schema.Resource {
+
+	return &schema.Resource{
+		Description: `Genesys Cloud outbound digitalruleset`,
+
+		CreateContext: provider.CreateWithPooledClient(createOutboundDigitalruleset),
+		ReadContext:   provider.ReadWithPooledClient(readOutboundDigitalruleset),
+		UpdateContext: provider.UpdateWithPooledClient(updateOutboundDigitalruleset),
+		DeleteContext: provider.DeleteWithPooledClient(deleteOutboundDigitalruleset),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			`name`: {
+				Description: `The name of the digital rule set`,
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			`version`: {
```

Comment:
We usually leave the version out of the resource schema and just set it ourselves in the code